src/main.cpp

```diff
@@ -962,7 +963,10 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
     if (pfMissingInputs)
         *pfMissingInputs = false;
 
-    if (!CheckTransaction(tx, state))
+    // Check for activation of rule changes
+    const bool protocol_cleanup = (chainActive.Tip()->GetMedianTimePast() >= Params().ProtocolCleanupActivationTime());
```

Comment:
I think it would be warranted to add an (inline maybe) function that returns the protocol cleanup state.

Reply:
My inclination is the same, but none of the other `CChainParams` methods work with block chain data structures. Maybe I could put an inline function in `main.h` however.

Reply:
Done.